From abb06e3c7a0153bc886e8cf0fd80631c29a2f951 Mon Sep 17 00:00:00 2001
From: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>
Date: Fri, 18 Mar 2022 08:42:43 -0700
Subject: [PATCH 13/13] configure: reduce libvpl version requirement

Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>
---
 configure | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure b/configure
index 1ddffd1..2199c20 100755
--- a/configure
+++ b/configure
@@ -6568,11 +6568,11 @@ elif enabled libmfx; then
 elif enabled libvpl; then
 # Consider pkg-config only. The name of libmfx is still passed to check_pkg_config function for --enable-libvpl option
 # because QSV has dependency on libmfx, we can use the same dependency if using libmfx in this check. The package name
-# is extracted from "vpl >= 2.6"
-    check_pkg_config libmfx "vpl >= 2.6" "mfxvideo.h mfxdispatcher.h" MFXLoad && \
-        warn "build FFmpeg against oneVPL 2.6+, OPAQUE memory, multi-frame encode, user plugins\n"\
+# is extracted from "vpl >= 2.5"
+    check_pkg_config libmfx "vpl >= 2.5" "mfxvideo.h mfxdispatcher.h" MFXLoad && \
+        warn "build FFmpeg against oneVPL 2.5+, OPAQUE memory, multi-frame encode, user plugins\n"\
              "and LA_EXT rate control mode in FFmpeg QSV won't be supported." ||
-            die "ERROR: libvpl >= 2.6 not found"
+            die "ERROR: libvpl >= 2.5 not found"
 
     check_cc MFX_EXTBUFF_AV1_BITSTREAM_PARAM "mfxdefs.h mfxstructures.h" "MFX_EXTBUFF_AV1_BITSTREAM_PARAM"
 fi
-- 
1.8.3.1

