From 35fd5b13b6c6377f820a4db35852081bd4fbf688 Mon Sep 17 00:00:00 2001
From: Haihao Xiang <haihao.xiang@intel.com>
Date: Mon, 6 Dec 2021 13:42:53 +0800
Subject: [PATCH 14/18] qsvenc: b_strategy can be applied to AV1 encoder

---
 libavcodec/qsvenc.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/libavcodec/qsvenc.c b/libavcodec/qsvenc.c
index c71fd81d54..c6de8246cf 100644
--- a/libavcodec/qsvenc.c
+++ b/libavcodec/qsvenc.c
@@ -924,6 +924,17 @@ static int init_video_param(AVCodecContext *avctx, QSVEncContext *q)
             q->extco2.Header.BufferId = MFX_EXTBUFF_CODING_OPTION2;
             q->extco2.Header.BufferSz = sizeof(q->extco2);
 
+            q->extparam_internal[q->nb_extparam_internal++] = (mfxExtBuffer *)&q->extco2;
+        } else if (avctx->codec_id == AV_CODEC_ID_AV1) {
+            // AV1e has better quality when enable B_PYRAMID, so
+            // enable it by default
+            q->extco2.BRefType = MFX_B_REF_PYRAMID;
+            if (q->b_strategy >= 0)
+                q->extco2.BRefType = q->b_strategy ? MFX_B_REF_PYRAMID : MFX_B_REF_OFF;
+
+            q->extco2.Header.BufferId = MFX_EXTBUFF_CODING_OPTION2;
+            q->extco2.Header.BufferSz = sizeof(q->extco2);
+
             q->extparam_internal[q->nb_extparam_internal++] = (mfxExtBuffer *)&q->extco2;
         }
 #endif
-- 
2.25.1

