From 51ac1a4093fc22d5b6b1e155c0a09941e16c3a3a Mon Sep 17 00:00:00 2001
From: Holy Wu <holywu@gmail.com>
Date: Thu, 9 Jul 2020 09:29:12 +0800
Subject: [PATCH 3/3] Add rev_target to libvmaf_rc for proper building order

---
 libvmaf/src/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libvmaf/src/meson.build b/libvmaf/src/meson.build
index 0fe9423..60bc2e2 100644
--- a/libvmaf/src/meson.build
+++ b/libvmaf/src/meson.build
@@ -199,7 +199,7 @@ libvmaf_rc_sources = [
 
 libvmaf_rc = both_libraries(
     'vmaf_rc',
-    libvmaf_rc_sources,
+    [libvmaf_rc_sources, rev_target],
     include_directories : [vmaf_include, libvmaf_inc],
     c_args : vmaf_cflags_common,
     cpp_args : vmaf_cflags_common,
-- 
1.8.3.1

